src/components/Home/News/newsTypes.ts

```diff
@@ -0,0 +1,49 @@
+import { Tag, Article } from "api/articles"
+
+/**
+ * Name of the event that has to be fired in order to update the preference
+ * of a tag (favourite or not) in the state of the NewsManager component
+ */
+export const UPDATE_PREFERENCE_EVENT_NAME = "update-preference"
+
+/** Enum to represent whether a tag is favourite or not */
+export enum Preference {
+    OTHER,
+    FAVOURITE,
+}
+
+export type TagWithData = Tag & {
+    /** The column in which the card has to be inserted */
+    column: "left" | "right"
+    /** The height of the card */
+    height: number
+    /** Whether the tag is favourite or not */
+    preference: Preference
+}
+
+/**
+ * Interface of an object that maps a tag name to its preference
+ *
+ * TODO: use tag id if and when there will be one
+ */
+export interface Preferences {
+    [key: Tag["name"]]: Preference
+}
```

Comment:
voglio dire, un enum con solo due valori è un bool alla fine no? ahah

cioè tienilo così se secondo te è più intuitivo in questo modo/è troppo complesso da modificare, però mi sembra facile l'idea di sostituire questa cosa con un campo `favourite: bool` direttamente, e puoi storare l'elenco dei tag preferiti come un array, tutti i tag che non sono nell'array sono automaticamente non preferiti.
Immagino in questo modo potresti filtrare tutti i tag in un modo tipo
```ts
tags.filter(tag => favourites.includes(tag))
tags.filter(tag => !favourites.includes(tag))
```
per distinguere i due casi, dove favourites mi immagino sia l'equivalente dell'attuale Preferences che starebbe nell'ipotetico context dell'altro commento idk ahah, vedi tu

Reply:
Si, all'inizio avevo un boolean, poi non mi ricordo esattamente il motivo ma mi era sembrato più pulito mettere un enum. Anche se alla fine non è che cambi più di tanto, quindi bo

Reply:
si beh a sto punto se devi fare il context decidi te come ti sembra più pulito scrivere questa cosa 👌